src/main/java/org/jenkinsci/plugins/pipeline/github/GitHubHelper.java

```diff
@@ -36,6 +36,20 @@ private GitHubHelper() {
         // go away
     }
 
+    public static Boolean isAuthorized(@Nonnull final Job<?,?> job, @Nonnull final String User) {
+        ExtendedGitHubClient client = getGitHubClient(job);
+        RepositoryId repository = getRepositoryId(job);
+        CollaboratorService collaboratorService = new CollaboratorService(client);
+
+        try {
+            return collaboratorService.isCollaborator(repository, User);
```

Comment:
This ends up calling this endpoint: 
```
/repos/${OWNER}/${REPO}/collaborators/${username}
```

So it leaves all of the decision logic back on github instead of here.